packages/kbn-openapi-bundler/src/bundler/bundle_document.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { RefResolver } from './ref_resolver';
+import { processDocument } from './process_document';
+import { createDereferenceConditionallyProcessor } from './document_processors/dereference_conditionally';
+import { createSkipNodeWithInternalPropProcessor } from './document_processors/skip_node_with_internal_prop';
+import { createModifyPartialProcessor } from './document_processors/modify_partial';
+import { createSkipInternalPathProcessor } from './document_processors/skip_internal_path';
+import { ResolvedDocument } from './types';
+import { createRemovePropsProcessor } from './document_processors/remove_props';
+import {
+  X_CODEGEN_ENABLED,
+  X_INTERNAL,
+  X_MODIFY,
+} from './document_processors/lib/known_custom_props';
+import { toAbsolutePath } from './lib/to_absolute_path';
+import { createModifyRequiredProcessor } from './document_processors/modify_required';
+
+export class SkipException extends Error {
+  constructor(public documentPath: string, message: string) {
+    super(message);
+  }
+}
+
+export async function bundleDocument(documentPath: string): Promise<ResolvedDocument> {
```

Comment:
I've added a simple comment. The main reason I refrained from adding a comment here because function's behavior defined by processors. I have an idea to move processors configuration out but it's to be discussed.